TypeCobol/Compiler/Diagnostics/CrossChecker.cs

```diff
@@ -204,6 +204,30 @@ public override bool Visit(Evaluate evaluate)
             return true; 
         }
 
+        public override bool Visit(Goback goback)
+        {
+            int gobackIndex = goback.Parent.ChildIndex(goback);
+            int lastIndex = goback.Parent.Children.Count - 1;
+            // last child is a period separator else there is a warning
+            if (gobackIndex < (lastIndex - 1))
+            {
+                DiagnosticUtils.AddError(goback.Parent.Children[gobackIndex + 1], "There should be no statement after \"GOBACK\"", MessageCode.Warning);
+            }
+            return true;
+        }
+
+        public override bool Visit(Stop stop)
+        {
+            int stopIndex = stop.Parent.ChildIndex(stop);
```

Comment:
Can you factorize the code between Stop and Goback ?

Reply:
done
https://github.com/TypeCobolTeam/TypeCobol/tree/65b1e7e18f92e701890f964b6dff1654306913ef